src/systems/isq/include/units/isq/electromagnetism.h

```diff
@@ -106,8 +106,9 @@ QUANTITY_SPEC(coupling_factor, mutual_inductance / pow<1, 2>(pow<2>(self_inducta
 QUANTITY_SPEC(leakage_factor, pow<2>(coupling_factor));
 QUANTITY_SPEC(conductivity, electric_current_density / electric_field_strength, quantity_character::scalar);
 QUANTITY_SPEC(resistivity, 1 / conductivity);
-QUANTITY_SPEC(power, voltage* electric_current);
-inline constexpr auto instantaneous_power = power;
+// QUANTITY_SPEC(power, voltage* electric_current);  // TODO conflicts with mechanical power
+// inline constexpr auto instantaneous_power = power;
```

Comment:
Below is what I currently have. As you know, Clang is prone to omit the template arguments in error messages.

Then there's most quantities from "ISO 80000-9:2019, Part 9: Physical chemistry and molecular physics". They will probably need to be parametrized on a structure describing a substance. That's to avoid adding apples and oranges, i.e. 10 electrons + 5 molecules of water, both terms which are values of the quantity number of entities.

```C++
struct context { };
inline constexpr context mechanics;
inline constexpr context thermodynamics;
inline constexpr context electromagnetism;
inline constexpr context light_and_optical_radiation;
inline constexpr context radiometry;
inline constexpr context acoustics;
inline constexpr context in_a_mixture;
inline constexpr context in_a_solution;
inline constexpr context pressure_basis;
inline constexpr context concentration_basis;
inline constexpr context atomic_physics;
inline constexpr context ionizing_radiation;

struct photometric_condition { };
inline constexpr photometric_condition photopic_vision = {};
inline constexpr photometric_condition scotopic_vision = {};
inline constexpr photometric_condition mesopic_vision  = {};

template<const context&> extern const quantity efficiency;
template<const context&> extern const quantity power;
template<const photometric_condition&> extern const quantity luminous_efficacy_of_radiation;
template<const photometric_condition&> extern const quantity maximum_luminous_efficacy;
template<const photometric_condition&> extern const quantity luminous_flux;
template<const context& = light_and_optical_radiation> extern const quantity reflectance;
template<const context& = light_and_optical_radiation> extern const quantity transmittance;
template<const context&> extern const quantity linear_attenuation_coefficient;
template<const context&> extern const quantity mass_attenuation_coefficient;
template<const context&> extern const quantity standard_absolute_activity;
template<const context&> extern const quantity equilibrium_constant;
template<const context&> extern const quantity magnetic_dipole_moment;
template<const context&> extern const quantity radiant_energy;

template<> inline constexpr auto power<mechanics> = defn<"𝘗", "𝘗 = 𝙁 ⋅ 𝙫", force("𝙁"), velocity("𝙫")>();

template<> inline constexpr auto efficiency<mechanics> = defn<"𝜂", "𝜂 = 𝘗ₒᵤₜ/𝘗ᵢₙ", power<mechanics>("𝘗")>();

template<> //
inline constexpr auto efficiency<thermodynamics> = defn<"𝜂", "𝜂 = 𝘞/𝘘", work("𝘞"), heat("𝘘")>();

template<> //
inline constexpr auto magnetic_dipole_moment<electromagnetism> = defn<"𝙟ₘ", "𝙟ₘ = 𝜇₀𝙢", magnetic_constant("𝜇₀"), magnetic_moment("𝙢")>();

template<> //
inline constexpr auto power<electromagnetism> = defn<"𝘱", "𝘱 = 𝘶𝘪", voltage("𝘶"), electric_current("𝘪")>();

template<> //
inline constexpr auto radiant_energy<electromagnetism> = defn<"𝘘ₑ", energy>();

template<const photometric_condition& C> constexpr auto luminous_efficiency          = defn<"𝘝", "𝘝 = 𝘒/𝘒ₘ", luminous_efficacy_of_radiation<C>("𝘒"), maximum_luminous_efficacy<C>("𝘒ₘ")>();
template<const photometric_condition& C> constexpr auto spectral_luminous_efficiency = defn<"𝘝(𝜆)", "dim 𝘝(𝜆) = 1">();

template<const photometric_condition& C> constexpr quantity luminous_efficacy_of_radiation = defn<"𝘒", "𝘒 = 𝛷ᵥ/𝛷ₑ", luminous_flux<C>("𝛷ᵥ"), radiant_flux("𝛷ₑ")>();
template<const photometric_condition& C> constexpr auto spectral_luminous_efficacy         = defn<"𝘒(𝜆)", "𝘒(𝜆) = 𝘒ₘ𝘝(𝜆)", maximum_luminous_efficacy<C>("𝘒ₘ"), spectral_luminous_efficiency<C>("𝘝(𝜆)"), wavelength("𝜆")>();
template<const photometric_condition& C> constexpr quantity maximum_luminous_efficacy      = defn<"𝘒ₘ", "𝘒ₘ = 683/𝘝(𝜆_{cd}) lm W⁻¹", spectral_luminous_efficiency<C>("𝘝(𝜆)"), "𝜆_{cd} = 540 × 10¹² Hz">();
template<const photometric_condition& C> constexpr auto luminous_efficacy_of_a_source      = defn<"𝜂ᵥ", "𝜂ᵥ = 𝛷ᵥ/𝘗", luminous_flux<C>("𝛷ᵥ"), power<mechanics>("𝘗")>();

template<const photometric_condition& C> constexpr auto luminous_energy    = defn<"𝘘ᵥ", "𝘘ᵥ = 𝘒ₘ∫₀^∞𝘘_{e,𝜆}(𝜆)𝘝(𝜆)d𝜆", maximum_luminous_efficacy<C>("𝘒ₘ"), spectral_radiant_energy("𝘘_{e,𝜆}(𝜆)"), spectral_luminous_efficiency<C>("𝘝(𝜆)"), wavelength("𝜆")>();
template<const photometric_condition& C> constexpr quantity luminous_flux  = defn<"𝛷ᵥ", "𝛷ᵥ = d𝘘ᵥ/d𝘵", luminous_energy<C>("𝘘ᵥ"), time("𝘵")>();
template<const photometric_condition& C> constexpr auto luminous_intensity = defn<"𝘐ᵥ", "𝘐ᵥ = d𝛷ᵥ/d𝛺", luminous_flux<C>("𝛷ᵥ"), solid_angle("𝛺")>();
template<const photometric_condition& C> constexpr auto luminance          = defn<"𝘓ᵥ", "𝘓ᵥ = (d𝘐ᵥ/d𝘈)(1/cos 𝛼)", luminous_intensity<C>("𝘐ᵥ"), area("𝘈")>();
template<const photometric_condition& C> constexpr auto illuminance        = defn<"𝘌ᵥ", "𝘌ᵥ = d𝛷ᵥ/d𝘈", luminous_flux<C>("𝛷ᵥ"), area("𝘈")>();
template<const photometric_condition& C> constexpr auto luminous_exitance  = defn<"𝘔ᵥ", "𝘔ᵥ = d𝛷ᵥ/d𝘈", luminous_flux<C>("𝛷ᵥ"), area("𝘈")>();
template<const photometric_condition& C> constexpr auto luminous_exposure  = defn<"𝘏ᵥ", "𝘏ᵥ = d𝘘ᵥ/d𝘈", luminous_energy<C>("𝘘ᵥ"), area("𝘈")>();

template<> inline constexpr auto reflectance<>   = defn<"𝜌", "𝜌 = 𝛷ᵣ/𝛷ₘ", radiant_flux("𝛷")>();
template<> inline constexpr auto transmittance<> = defn<"𝜏", "𝜏 = 𝛷ₜ/𝛷ₘ", radiant_flux("𝛷")>();

template<> //
constexpr auto& linear_attenuation_coefficient<radiometry> = linear_extinction_coefficient;
template<> //
inline constexpr auto mass_attenuation_coefficient<radiometry> = defn<"𝜇ₘ", "dim 𝜇ₘ = 𝜇𝜌⁻¹", linear_attenuation_coefficient<radiometry>("𝜇"), mass_density("𝜌_𝘈")>();

template<fixed_string R = "20 µPa"> constexpr auto sound_pressure_level     = defn<"𝘓_𝘱", "𝘓_𝘱 = 10 lg(𝘱²/𝘱₀²) dB", R("𝘱₀ = @"), sound_pressure("𝘱")>();
template<fixed_string R = "1 pW"> constexpr auto sound_power_level          = defn<"𝘓_𝘞", "𝘓_𝘞 = 10 lg(𝘗/𝘗₀) dB", R("𝘗₀ = @"), sound_power("𝘗")>();
template<fixed_string R = "400 µPa² s"> constexpr auto sound_exposure_level = defn<"𝘓_𝘌", "𝘓_𝘌 = 10 lg(𝘌/𝘌₀) dB", R("𝘌₀ = @"), sound_exposure("𝘌")>();

template<> inline constexpr auto reflectance<acoustics>    = defn<"𝘳", "dim 𝘳 = 1">();
template<> inline constexpr auto transmittance<acoustics>  = defn<"𝜏", "dim 𝜏 = 1">();

template<> inline constexpr auto standard_absolute_activity<in_a_mixture> = defn<"𝜆_X^⊝", "𝜆_X^⊝ = 𝜆 ⃰_X(𝘱^⊝)", "𝘱^⊝ = 10⁵ Pa", absolute_activity("𝜆_X"), pressure("𝘱")>();

template<> inline constexpr auto standard_absolute_activity<in_a_solution> = defn<"𝜆_B^⊝", "𝜆_B^⊝ = lim_{∑𝘣_B→0}[𝜆_B((𝘱^⊝)𝘣^⊝/𝘣_B)]", "𝘱^⊝ = 10⁵ Pa", molality("𝘣"), absolute_activity("𝜆_B"), pressure("𝘱")>();

template<> inline constexpr auto equilibrium_constant<pressure_basis>      = defn<"𝘒_𝘱", "𝘒_𝘱 = ∏_B(𝘱_B)^{𝘷_B}", partial_pressure("𝘱"), stoichiometric_number_of_substance("𝘷_B")>();
template<> inline constexpr auto equilibrium_constant<concentration_basis> = defn<"𝘒_𝘤", "𝘒_𝘤 = ∏_B(𝘤_B)^{𝘷_B}", particle_concentration("𝘤"), stoichiometric_number_of_substance("𝘷_B")>();

template<> inline constexpr auto magnetic_dipole_moment<atomic_physics> = defn<"", "Δ𝘞 = -𝞵 ⋅ 𝘽", energy("Δ𝘞"), magnetic_flux_density("𝘽")>();

template<> inline constexpr auto radiant_energy<ionizing_radiation>      = defn<"", "𝘙", energy("𝘌")>();

template<> //
inline constexpr auto linear_attenuation_coefficient<ionizing_radiation>                     = defn<"", "𝜇 = d𝘯/d𝘭 = (1/𝘕)(d𝘕/d𝘭)", number_of_entities(""), distance("")>();
template<> //
inline constexpr auto mass_attenuation_coefficient<ionizing_radiation>                       = defn<"", "𝜇_𝘮 = 𝜇/𝜌", linear_attenuation_coefficient<ionizing_radiation>(""), mass_density("𝜌")>();
```

Reply:
This means that SI base quantities `luminous_intensity` and `amount_of_substance` are parametrized, and not one specialization is more "base" than the other.